master/internal/task/allocation.go

```diff
@@ -353,6 +355,7 @@ func (a *Allocation) ResourcesAllocated(ctx *actor.Context, msg sproto.Resources
 			IsMultiAgent: len(a.reservations) > 1,
 		})
 	}
+	a.reservationsStarted = true
```

Comment:
Let's say
1. in line 352 here,`r.Start`, tells agent actor to start a task
2. we set reservationStarted here.
3. agent network connection breaks and it cannot write message # 1

then I suspect in the current code we'll *probably* receive `ContainerStopped` when agent actor will die one way or another?... is that right?

Reply:
yeah, that is basically my thought exactly

Reply:
so long as we dont fail without handling it correctly, other downstream failures should cause us to handle this correctly